### PR TITLE
Feature/use header enums

### PR DESCRIPTION
**Pull request by wow-such-code**

### CODE REVIEW

File: src/main/java/life/qbic/expdesign/io/QBiCDesignReader.java [hunk 1]
```diff
@@ -25,15 +27,15 @@
...
+      "EXPERIMENT", "SAMPLE TYPE", OpenbisPropertyCodes.Q_SECONDARY_NAME.name(), "PARENT"));
+  private List<String> entityMandatory = new ArrayList<>(Arrays.asList(OpenbisPropertyCodes.Q_NCBI_ORGANISM.name()));
+  private List<String> extractMandatory = new ArrayList<>(Arrays.asList(OpenbisPropertyCodes.Q_PRIMARY_TISSUE.name()));
+  private List<String> extractSpecials = new ArrayList<>(Arrays.asList(OpenbisPropertyCodes.Q_TISSUE_DETAILED.name()));
+  private List<String> testMandatory = new ArrayList<>(Arrays.asList(OpenbisPropertyCodes.Q_SAMPLE_TYPE.name()));
+  private List<String> mhcSpecials = new ArrayList<>(Arrays.asList(OpenbisPropertyCodes.Q_MHC_CLASS.name()));
   private List<String> sampleTypesInOrder =
-      new ArrayList<String>(Arrays.asList("Q_BIOLOGICAL_ENTITY", "Q_BIOLOGICAL_SAMPLE",
+      new ArrayList<>(Arrays.asList("Q_BIOLOGICAL_ENTITY", "Q_BIOLOGICAL_SAMPLE",
           "Q_TEST_SAMPLE", "Q_MHC_LIGAND_EXTRACT", "Q_NGS_SINGLE_SAMPLE_RUN", "Q_MS_RUN"));
```
Comment (jenniferboedker) [inline src/main/java/life/qbic/expdesign/io/QBiCDesignReader.java:40]: Can the content here (the codes) also be replaced?

File: src/main/java/life/qbic/expdesign/io/MHCLigandDesignReader.java [hunk 2]
```diff
@@ -97,58 +97,73 @@ public String[] getMHCClass(String antibody) {
...
   private void fillParsedCategoriesToValuesForRow(Map<String, Integer> headerMapping,
       String[] row) {
     // logger.info("Collecting possible CV entries for row.");
-    addValueForCategory(headerMapping, row, "Organism");
-    addValueForCategory(headerMapping, row, "Tissue");
-    addValueForCategory(headerMapping, row, "Dignity");
-    addValueForCategory(headerMapping, row, "Tumor Type");
-    addValueForCategory(headerMapping, row, "TNM");
-    addValueForCategory(headerMapping, row, "Antibody");
-    addValueForCategory(headerMapping, row, "MHC Class");
```
Comment (jenniferboedker) [inline src/main/java/life/qbic/expdesign/io/MHCLigandDesignReader.java:151]: did you remove some of these values on purpose?

**Review by jenniferboedker — CHANGES_REQUESTED** [review]
See the comments I already send :)

Great, that you also improved the code along your change 👍🏼

File: src/main/java/life/qbic/expdesign/io/MHCLigandDesignReader.java (hunk 2 above)
Comment (wow-such-code) [reply src/main/java/life/qbic/expdesign/io/MHCLigandDesignReader.java:151]: Yes, I noticed many of them are actually not controlled by vocabularies and Antibody has a check in the parser itself, so it only makes sense to provide a check for Organism, Tissue and Tumor Type to the wizard.

File: src/main/java/life/qbic/expdesign/io/QBiCDesignReader.java (hunk 1 above)
Comment (wow-such-code) [reply src/main/java/life/qbic/expdesign/io/QBiCDesignReader.java]: indeed, good idea

**Review by jenniferboedker — APPROVED** [review]

**Pull request merged**